waku/waku_rln_relay/rln_relay.nim

```diff
@@ -241,19 +240,39 @@ proc validateMessage*(rlnPeer: WakuRLNRelay,
   if hasDup.isErr():
     waku_rln_errors_total.inc(labelValues=["duplicate_check"])
   elif hasDup.value == true:
-    debug "invalid message: message is spam", payload = string.fromBytes(msg.payload)
+    debug "invalid message: message is spam", payloadLen = msg.payload.len
     waku_rln_spam_messages_total.inc()
     return MessageValidationResult.Spam
 
-  # insert the message to the log
-  # the result of `updateLog` is discarded because message insertion is guaranteed by the implementation i.e.,
-  # it will never error out
-  discard rlnPeer.updateLog(proofMetadataRes.get())
-  debug "message is valid", payload = string.fromBytes(msg.payload)
+  debug "message is valid", payloadLen = msg.payload.len
   let rootIndex = rlnPeer.groupManager.indexOfRoot(proof.merkleRoot)
   waku_rln_valid_messages_total.observe(rootIndex.toFloat())
   return MessageValidationResult.Valid
 
+proc validateMessageAndUpdateLog*(
```

Comment:
@rymnc created this function, which is like the old `validateMessage` but now we have:
* validateMessage
* validateMessageAndUpdateLog